src/kvirc/kernel/KviIrcConnection.cpp

```diff
@@ -832,14 +832,12 @@ bool KviIrcConnection::sendData(const char * pcBuffer, int iBuflen)
 			m_pConsole->outputNoFmt(KVI_OUT_SOCKETWARNING, __tr2qs("[LINK WARNING]: Socket message truncated to 512 bytes."));
 	}
 
-	KviDataBuffer * pData = new KviDataBuffer(iBuflen + 3);
+	KviDataBuffer * pData = new KviDataBuffer(iBuflen + 2);
 	KviMemory::move(pData->data(), pcBuffer, iBuflen);
 	*(pData->data() + iBuflen) = '\r';
 	*(pData->data() + iBuflen + 1) = '\n';
-	*(pData->data() + iBuflen + 2) = '\0';
 
-	QString szMsg = (const char *)(pData->data());
-	szMsg.truncate(iBuflen);
+	QString szMsg = QString::fromUtf8((const char *)(pData->data()), iBuflen);
```

Comment:
Is it guaranteed to be utf8?

Reply:
It's a raw string (byte sequence) coming from the user (/quote, /raw), we can't be 100% sure of its encoding.
The  implicit ctor used before internally called QString::fromUtf8(), so there's no change in behavior: https://doc.qt.io/qt-5/qstring.html#QString-7